### PR TITLE
secretstorage init at version 2.3.1 

### PR DESCRIPTION
**Pull request by teto**
###### Motivation for this change

A program I use (i3pystatus) has modules depending on this python module (via the keyring python module).
https://pypi.python.org/pypi/SecretStorage

I've tested it through:
`nix-shell -p python36Packages.keyring -p python36Packages.secretstorage -I ~/nixpkgs --command "keyring set gmail password"`
This is my first PR to nixpkgs so this might need more work. I'll try to look into the sandboxing thing.

###### Things done

- [ ] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] Linux
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/development/python-modules/secretstorage/default.nix [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    repo = "secretstorage";
+    rev = "2636a47b45aff51a21dfaf1cbd9f1f3c1347f7f4";
+    sha256 = "1xs35ywqxxriwhm79kvivad0hszcm4ah29v34vbshlqvc66qr171";
+  };
+
+  postPatch = "patchShebangs .";
+  propagatedBuildInputs = [ dbus-python cryptography ];
+  doCheck = false;
+
+  meta = with stdenv.lib; {
```
Comment (vyp) [inline pkgs/development/python-modules/secretstorage/default.nix]: Did you want to add yourself as maintainer?

Comment (teto) [reply pkgs/development/python-modules/secretstorage/default.nix]: not really. I am not proficient with nixos nor I am knowledgeable about dbus/keyring stuff. If it's mandatory, I can put myself as a best effort thing.

Comment (vyp) [reply pkgs/development/python-modules/secretstorage/default.nix]: I think it is mandatory, but maybe someone else will want to be a maintainer. However, it shouldn't be too hard to be a maintainer, it should mainly just be about updating the revision and sha256 whenever there's a new release, and checking to see if it builds/works.

File: pkgs/development/python-modules/secretstorage/default.nix [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+buildPythonPackage rec {
+  pname = "secretstorage";
+  version = "2.3.1";
+  name = "${pname}-${version}";
+
+  src = pkgs.fetchFromGitHub {
+    owner = "mitya57";
+    repo = "secretstorage";
+    rev = "2636a47b45aff51a21dfaf1cbd9f1f3c1347f7f4";
```
Comment (vyp) [inline pkgs/development/python-modules/secretstorage/default.nix]: This should just be `"${version}"`. And note the sha256 will therefore also change.

File: pkgs/development/python-modules/secretstorage/default.nix [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+  name = "${pname}-${version}";
+
+  src = fetchFromGitHub {
+    owner = "mitya57";
+    repo = "secretstorage";
+    rev = version;
+    sha256 = "1sjd2jjbxgkkxyrfwx89x0hsnn39w2cr2qkxbg1iz52znr4sqism";
+  };
+
+  postPatch = "patchShebangs .";
```
Comment (FRidh) [inline pkgs/development/python-modules/secretstorage/default.nix]: why is this needed?

Comment (teto) [reply pkgs/development/python-modules/secretstorage/default.nix]: this is a leftover from the derivation I took inspiration from. It should not be necessary.

**Comment by FRidh** [comment]
Thank you. Added it to #29009.
